tests/test_database_sqlite.py

```diff
@@ -1,115 +1,116 @@
     """A async database test class.
 
     Test the async methods of the opsdroid sqlite database class.
 
     """
 
-    async def test_connect(self):
-        """Test database connection.
-
-        This method will test the database connection of sqlite database.
-        As the database is created `opsdroid` table is created first.
-
-        """
-        database = DatabaseSqlite({"path": "sqlite.db"})
-        opsdroid = amock.CoroutineMock()
-        opsdroid.eventloop = self.loop
-
-        try:
-            await database.connect()
-            table = database.table
-            client = type(database.client).__name__
-            await database.disconnect()
-        except NotImplementedError:
-            raise Exception
-        else:
-            self.assertEqual("opsdroid", table)
-            self.assertEqual("Connection", client)
-
-    async def test_disconnect(self):
-        """Test of database disconnection.
-
-        This method will test the database disconnection of sqlite database.
-
-        """
-        database = DatabaseSqlite({"path": "sqlite.db"})
-        opsdroid = amock.CoroutineMock()
-        opsdroid.eventloop = self.loop
-
-        try:
-            await database.connect()
-            await database.disconnect()
-        except NotImplementedError:
-            raise Exception
-        else:
-            pass
-
-    async def test_get_put_and_delete(self):
-        """Test get, put and delete functions of database
-
-        This method will test the get, put and delete functions which help to read
-        and write data from the database. The function `put` a value with
-        key and asserts the same value after the `get` operation is completed
-        followed by the `delete` operation which deletes the key.
-
-        """
-        database = DatabaseSqlite({"path": "sqlite.db"})
-        opsdroid = amock.CoroutineMock()
-        opsdroid.eventloop = self.loop
-
-        try:
-            await database.connect()
-            table = database.table
-            client = type(database.client).__name__
-            await database.put("hello", {})
-            data = await database.get("hello")
-            await database.delete("hello")
-            await database.disconnect()
-        except NotImplementedError:
-            raise Exception
-        else:
-            self.assertEqual("opsdroid", table)
-            self.assertEqual({}, data)
-            self.assertEqual("Connection", client)
-
-    async def test_deprecated_path(self):
-        database = DatabaseSqlite({"file": "sqlite.db"})
-        assert database.db_file == "sqlite.db"
+
+@pytest.mark.asyncio
+async def test_connect():
+    """Test database connection.
+
+    This method will test the database connection of sqlite database.
+    As the database is created `opsdroid` table is created first.
+
+    """
+    database = DatabaseSqlite({"path": "sqlite.db"})
+    opsdroid = amock.CoroutineMock()
```

Comment:
We have an `opsdroid` fixture, I believe if you add opsdroid as a parameter of the function you could then remove this line.

So this function will look like: `async def test_connect(opsdroid):`
I'm happy to keep it as it is if you think is best (or if the fixture won't work haha)

If it works and you are happy with it, then we could refactor the tests a bit more and add opsdroid as a parameter to every function on this file.

Reply:
I tried this but unfortunately could not get it to work, I kept getting an error about the 'opsdroid' fixture not being found.